hazelcast/src/main/java/com/hazelcast/nio/NodeIOService.java

```diff
@@ -231,7 +231,20 @@ public boolean getSocketNoDelay() {
     }
 
     @Override
-    public int getSelectorThreadCount() {
+    public int getInputSelectorThreadCount() {
+        int count = node.getGroupProperties().IO_INPUT_THREAD_COUNT.getInteger();
+        if (count != -1) {
```

Comment:
It should not be done here. Group properties should be fixed so they have build in validation. This should not be added at every location you need to use the property.